cdk/lib/frontend.ts

```diff
@@ -18,5 +34,360 @@ export class Frontend extends GuStack {
+        ],
+      }),
+    ];
+
+    const ec2App = new GuEc2App(this, {
+      applicationPort: 9000,
+      app: "frontend",
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: {
+        [Stage.CODE]: {
+          domainName:
+            "support.code.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+        [Stage.PROD]: {
+          domainName:
+            "support.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData,
+      roleConfiguration: {
+        additionalPolicies: policies,
+      },
+      scaling: {
+        PROD: {
+          minimumInstances: 3,
+          maximumInstances: 6,
+        },
+        CODE: {
+          minimumInstances: 1,
+          maximumInstances: 2,
+        },
+      },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+    });
+
+    // TODO: remove this tag after the migration
+    const ec2AppAsg = ec2App.autoScalingGroup;
+    Tags.of(ec2AppAsg).add("gu:riffraff:new-asg", "true");
+
+    // ---- Alarms ---- //
+
+    const alarmName = (shortDescription: string) =>
+      `URGENT 9-5 - ${this.stage} ${shortDescription}`;
+
+    const alarmDescription = (description: string) =>
+      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;
+
+    const alarmEnabledInProd = this.withStageDependentValue({
+      app,
+      variableName: "AlarmEnabledInProd",
+      stageValues: {
+        [Stage.CODE]: false,
+        [Stage.PROD]: true,
+      },
+    });
+
+    new GuAlarm(this, "NoHealthyInstancesAlarm", {
+      app,
+      alarmName: alarmName("no healthy instances for support-frontend"),
+      alarmDescription: alarmDescription(
+        "Cannot sell any subscriptions or contributions products"
+      ),
+      actionsEnabled: alarmEnabledInProd,
+      threshold: 0.5,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HealthyHostCount",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: ec2App.loadBalancer.loadBalancerFullName,
+          TargetGroup: ec2App.targetGroup.targetGroupFullName,
+        },
+        statistic: "Average",
+        period: Duration.seconds(60),
+      }),
+      snsTopicName: "reader-revenue-dev",
+    });
+
+    new GuAlarm(this, "ReducedHealthyInstancesAlarm", {
+      app,
+      alarmName: alarmName(
+        "reduced number healthy instances for support-frontend"
+      ),
+      alarmDescription: alarmDescription(
+        "Imminent issue cannot sell any subscriptions or contributions products"
+      ),
+      actionsEnabled: alarmEnabledInProd,
+      threshold: 2,
```

Comment:
I presume that this threshold is set to 2 because you want to run [at least 3 instances at all times in `PROD`](https://github.com/guardian/support-frontend/blob/c3c8289e84a301a72864ddf55cbddaa765594b48/cdk/lib/frontend.ts#L158). Now that you're defining all of this in TS, you could encode that logic (i.e. threshold = `minimumProdInstances - 1`), which means that this will continue working as desired if you ever scale up.

Reply:
Great suggestion! I've pushed that up